docs/release-info/release-notes.md

```diff
@@ -109,11 +523,47 @@ For more information, see [Migration guide: front-coded dictionaries](./migr-fro
 
 If you're already using this feature, you don't need to take any action. 
 
-
 ### Incompatible changes
 
+#### Removed the scan query legacy mode
+
+The native scan query legacy mode has been removed. It was introduced in Druid 0.11 to maintain compatibility during an upgrade from older versions of Druid where the scan query was part of a `contrib` extension.
+
+[#16659](https://github.com/apache/druid/pull/16659)
+
+Hard-coded `"legacy":false` following removal of the legacy mode to prevent error during rolling upgrades or downgrades.
+
+[#16793](https://github.com/apache/druid/pull/16793)
+
+#### ZK-based segment loading
+
+ZK-based segment loading is now disabled. ZK `servedSegmentsPath` was deprecated in Druid 0.7.1. This legacy path has been replaced by `liveSegmentsPath`.
+
+Segment-serving processes such as Peons, Historicals and Indexers no longer create ZK `loadQueuePath` entries. The `druid.zk.paths.loadQueuePath` and `druid.zk.paths.servedSegmentsPath` properties are no longer used.
+
+Move to HTTP-based segment loading first and then perform the version upgrade.
+
 ### Developer notes
 
+- Fixed the `hibernate-validator` warning displayed during maven build [#16746](https://github.com/apache/druid/pull/16746)
+- Replaced the `StorageAdapter` interface with the `CursorHolder` interface [#17024](https://github.com/apache/druid/pull/17024)
+
 #### Dependency updates
 
-The following dependencies have had their versions bumped:
\ No newline at end of file
+Bumped the versions of the following dependencies:
+
+- Apache Calcite to 1.37.0 [#16621](https://github.com/apache/druid/pull/16621)
+- Delta Kernel from 3.1.0 to 3.2.0 [#16513](https://github.com/apache/druid/pull/16513)
```

Comment:
This change is superseded by https://github.com/apache/druid/pull/17179, so it should say 3.2.1 instead of 3.2.0